components/organisms/InsightPage/InsightPage.tsx

```diff
@@ -552,6 +552,7 @@ const InsightPage = ({ edit, insight, pageRepos }: InsightPageProps) => {
             placeholder="Repository URL or Full Name (ex: open-sauced/open-sauced)"
             className="!w-full text-md text-gra"
             name={"query"}
+            value={repoSearchTerm}
```

Comment:
Most likely this is still not clearing as reported by @BekahHW  because the state is not being reset, i.e. `setRepoSearchTerm("")`. I see there are calls to this already, but maybe one is missing for this new use case of the search being a [controlled component](https://react.dev/learn/sharing-state-between-components#controlled-and-uncontrolled-components) now.

Reply:
You @nickytonline  are right search component is now a controlled component.

And I think `loadAndAddRepo` is handling the use case which is `setRepoSearchTerm("")` . Please let me know if I misunderstood something.

``` 
 const loadAndAddRepo = async (repoToAdd: string, isAddedFromCart = false) => {
    setAddRepoError(RepoLookupError.Initial);


    const hasRepo = repos.find((repo) => `${repo.full_name}` === repoToAdd);


    if (hasRepo) {
      return;
    }


    const { apiPaths, isValidUrl } = generateRepoParts(repoToAdd);


    if (!isValidUrl) {
      setAddRepoError(RepoLookupError.Invalid);
      return;
    }


    const { repoFullName } = apiPaths;


    setAddRepoLoading({ repoName: repoToAdd, isAddedFromCart, isLoading: true });
    try {
      const response = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/repos/${repoFullName}`);
      setAddRepoLoading({ repoName: repoToAdd, isAddedFromCart, isLoading: false });


      if (response.ok) {
        const addedRepo = (await response.json()) as DbRepo;


        setRepos((repos) => {
          return [...repos, addedRepo];
        });
        setAddRepoError(RepoLookupError.Initial);
        setRepoSearchTerm("");
      } else {
        const publicRepoResponse = await fetch(`https://api.github.com/repos/${repoFullName}`);


        if (publicRepoResponse.ok) {
          const publicRepo = await publicRepoResponse.json();


          // create a stub repo to send to API
          const addedRepo = {
            id: publicRepo.id,
            full_name: publicRepo.full_name,
          } as DbRepo;


          setRepos((repos) => {
            return [...repos, addedRepo];
          });
          setAddRepoError(RepoLookupError.Initial);
          setRepoSearchTerm("");
        } else {
          setAddRepoError(RepoLookupError.Invalid);
        }
      }
    } catch {
      setAddRepoLoading({ repoName: repoToAdd, isAddedFromCart, isLoading: false });
      setAddRepoError(RepoLookupError.Error);
    }
  };


  const handleAddRepository = async () => {
    await loadAndAddRepo(repoSearchTerm);
  };
```

[Link to code](https://github.com/open-sauced/app/blob/75992f53dc389709f26de88f3ad3a335cdc1ac51/components/organisms/InsightPage/InsightPage.tsx#L271C2-L331C5)

Reply:
@nickytonline waiting for your response 😅